tuf/repository_lib.py

```diff
@@ -1521,6 +1521,28 @@ def _generate_targets_fileinfo(target_files, targets_directory,
 
 
 
+def _get_hashes_and_length_if_needed(use_length, use_hashes, full_file_path,
+    storage_backend):
+  """
+  Calculate length and hashes only if they are required.
+  """
+
+  length = None
+  hashes = None
+  # We want to make sure we are calculating length and hashes only when
+  # they are needed. Otherwise, for adoptors of tuf with
+  # lots of delegations, this will cause unnecessary overhead.
```

Comment:
NIT: this is a more comprehensive version of the description in the docstring, perhaps we should merge them?